### ISSUE TITLE
MNE-Python channel unit warnings should be suppressed

### ISSUE THREAD

**Issue by jo780-full** [mne-tools/mne-bids#1100]
The original issue was resolved, but a second issue emerged on this thread. Skip to [this comment](https://github.com/mne-tools/mne-bids/issues/1100#issuecomment-1315893527) and read on from there. I've hidden the original problem in "details".  (signed, @drammock)

<details>

### Description of the problem

Hello, I am new to mne and have a problem with this dataset.

It shows 
ValueError: invalid literal for int() with base 10: ''

### Steps to reproduce

```Python
`datatype='eeg'
subject='024'
task='med1breath'
suffix='eeg'
bids_path=BIDSPath(subject=subject,task=task,suffix=suffix,datatype=datatype,root=bids_root)
sidecar_path=bids_path.copy().update(extension='.json')
raw = read_raw_bids(bids_path=bids_path)
`
```

### Link to data

[doi:10.18112/openneuro.ds003969.v1.0.0](https://doi.org/10.18112/openneuro.ds003969.v1.0.0)]

### Expected results

Reads file

### Actual results

![image](https://user-images.githubusercontent.com/84071740/201730018-600f2071-673b-480b-a8d7-f2b8c2d3be46.png)

### Additional information

Platform:         Windows-10-10.0.22000-SP0
Python:           3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]
Executable:       C:\Users\josep\anaconda3\python.exe
CPU:              AMD64 Family 23 Model 96 Stepping 1, AuthenticAMD: 16 cores
Memory:           15.4 GB

mne:              1.2.2
numpy:            1.20.3 {MKL 2021.4-Product with 8 threads}
scipy:            1.7.1
matplotlib:       3.4.3 {backend=module://matplotlib_inline.backend_inline}

sklearn:          0.24.2
numba:            0.54.1
nibabel:          Not found
nilearn:          Not found
dipy:             Not found
openmeeg:         Not found
cupy:             Not found
pandas:           1.3.4
pyvista:          Not found
pyvistaqt:        Not found
ipyvtklink:       Not found
vtk:              Not found
qtpy:             1.10.0 {PyQt5=5.9.7}
ipympl:           Not found
pyqtgraph:        Not found
pooch:            v1.6.0

mne_bids:         0.11.1
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   Not found
mne_connectivity: Not found
mne_icalabel:     Not found

</details>

**Comment by drammock** [comment]
> This issue should actually be transferred over to MNE-BIDS.

done

**Comment by sappelhoff** [comment]
Should probably be fixed here:

https://github.com/mne-tools/mne-bids/blob/0b1a61d7660b35438b78898878ef0c8b51d8716b/mne_bids/read.py#L621

**Comment by hoechenberger** [comment]
It's now easy to suppress these warnings in MNE, and that change will be included in MNE 1.4.

One MNE 1.5 (!) is out, we can safely switch over to that new functionality, as we intend to support the latest two versions of MNE in MNE-BIDS
